src/sage/graphs/domination.py

```diff
@@ -730,7 +737,7 @@ def _aux_with_rep(H, to_dom, u_next):
                 break
 
 
-def minimal_dominating_sets(G, to_dominate=None, work_on_copy=False, k=1):
+def minimal_dominating_sets(G, to_dominate=None, work_on_copy=True, k=1):
     r"""
```

Comment:
Should there be a deprecation period for this change of default?

Reply:
We could add a deprecation, but here I consider that we fix a bug. The default value of parameter ``work_on_copy`` was not consistent with the code: the graph was modified when set to ``True`` and a copy was created when set to ``False``.

Furthermore, I think it is an error to modify the input graph without any explicitly action from the user.

Reply:
OK, makes sense.